### ISSUE TITLE
Passing Unicode string as url to curl causes crash

### ISSUE THREAD

**Issue by stevvooe** [discoproject/disco#52]
Basically, one part passes a unicode string to setopt in pycurl. I recommend removing pycurl as dependency for disco. Traceback is below:

```
URL: u'http://localhost:8989/localhost/77/disco_tut@1260921574/reduce-index.txt'
Traceback (most recent call last):
  File "count_words.py", line 24, in <module>
    for word, total in result_iterator(results):
  File "...disco-0.2.3/pydisco/disco/core.py", line 466, in result_iterator
    sze, fd = comm.open_remote(url)
  File ".../disco-0.2.3/pydisco/disco/comm_curl.py", line 149, in open_remote
    conn = CurlConn(url, handle = c, expect = expect)
  File ".../disco-0.2.3/pydisco/disco/comm_curl.py", line 66, in __init__
    self.init_handle(url)
  File ".../disco-0.2.3/pydisco/disco/comm_curl.py", line 89, in init_handle
    self.handle.setopt(URL, url)
TypeError: invalid arguments to setopt
```

**Comment by tuulos** [comment]
A safe workaround should be to cast Unicode objects to plain strings before passing them to any Disco functions. Let's see if there some sane way to deal with this issue internally.

**Comment by stevvooe** [comment]
That might be of dubious safety. Though unlikely, just casting unicode to strings can cause problems. Do you really need pycurl as a dependency? Are there features that pycurl provides (other than performance) that are needed? I have no opinion either way, but its just something to think about. 

Keep up the good work!

**Comment by tuulos** [comment]
Cast unicode urls to strings in comm_curl. Closed by da2b41d0496f411622a05c332b00af57aec18395
